lib/bridge/IrcHandler.js

```diff
@@ -382,18 +382,22 @@ IrcHandler.prototype._serviceTopicQueue = Promise.coroutine(function*(item) {
             );
             return Promise.resolve();
         }
-        return this.ircBridge.getAppServiceBridge().getIntent().setRoomTopic(
-            entry.matrix.getId(),
-            item.topic
-        ).then(
+        return this.ircBridge.getAppServiceBridge().getIntent(
+            item.matrixUser.getId()
+        ).setRoomTopic(
+            entry.matrix.getId(), item.topic
+        ).catch(() => {
+            // Setter might not have powerlevels, trying again.
+            return this.ircBridge.getAppServiceBridge().getIntent()
+                .setRoomTopic(entry.matrix.getId(), item.topic);
+        }).then(
             () => {
                 entry.matrix.topic = item.topic;
                 return this.ircBridge.getStore().upsertRoomStoreEntry(entry);
             },
             (err) => {
                 item.req.log.error(`Error storing room ${entry.matrix.getId()} (${err.message})`);
-            }
-        );
+        });
```

Comment:
personally I'm more of a fan of having the curly on it's own line so it lines up nicely